userspace/libsinsp/plugin_filtercheck.cpp

```diff
@@ -127,6 +127,7 @@ std::unique_ptr<sinsp_filter_check> sinsp_filter_check_plugin::allocate_new() {
 
 bool sinsp_filter_check_plugin::extract_nocache(sinsp_evt* evt,
                                                 std::vector<extract_value_t>& values,
+                                                std::vector<extract_offset_t>* offsets,
```

Comment:
Do we really need a vector here considering we only return a single offset?
Or do we want to support extracting multiple offsets just like we support extracting multiple values?

Reply:
I'm not sure. It seems odd to have a vector for fields and a scalar for offsets. Would it make sense to make them both scalar everywhere?

Reply:
I think a single `extract` might push multiple values, that's the point of the API: https://github.com/falcosecurity/plugin-sdk-cpp/blob/d7cfeb72b07dcab3ad5bd78f3f6b8e07d1196a43/include/falcosecurity/internal/deps/plugin_types.h#L159C4-L168C97

I'd expect the same to be true for the offsets (ie: one should return an offset for each extracted value).

Reply:
I think the function signature is ok, but the actual implementation does not handle the case when there're multiple offsets. @FedeDP is this what you meant?

If yes, I agree. If we extract N values, we should also provide N offsets. 

@geraldcombs does it make sense for you too?

Reply:
cc @geraldcombs

Reply:
It makes sense to me. I'll take a look.